### DISCUSSION TITLE
Maybe someone could provide a tutorial for me?

### DISCUSSION THREAD

**Discussion by clpoz** [scripthookvdotnet/scripthookvdotnet#1448] [Q&A] [answered]
Hi every developers，I would like to create my own gtav environment for experiments，But I am not familiar about SHVDN，I coding with python and java well but I never  used C# for any .net development，I'm sincerely to see you give me a link to a tutorial, such as documentation and some sample instructions to help me get started with SHVDN，thank you very much!

**Comment by clpoz** [comment]
For example, after a long time of fumbling around, I just found out that after placing the .asi and .dll files in the root directory of the game, I also need to put the C# script code in the scripts directory where it was created, so that the game can load these scripts. I'm new to .net and c# related development, so I'm sincerely grateful to get an introductory tutorial on SHVDN

**Comment by kagikn** [comment, marked as answer]
Unfortunately, we haven't provided an introductory tutorial on SHVDN, but maybe you could try [our wiki](https://github.com/scripthookvdotnet/scripthookvdotnet/wiki) for some basic knowledge on SHVDN. However, I believe you should get used to C# first. For example, you can try [C# overview on Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/overview). Do note that you have to build scripts against .NET Framework 4.8 (or later) in SHVDN, not .NET 5+. Therefore, you can't use many new features such as `Span` or interfaces with default implementations (the latter needs dedicated runtime support even).
